src/native/libs/System.Security.Cryptography.Native/pal_evp_cipher.c

```diff
@@ -392,13 +392,27 @@ const EVP_CIPHER* CryptoNative_EvpDes3Cbc(void)
 const EVP_CIPHER* CryptoNative_EvpRC2Ecb(void)
 {
     // No error queue impact.
-    return EVP_rc2_ecb();
+#if HAVE_OPENSSL_EVP_RC2_ECB
+    if (API_EXISTS(EVP_rc2_ecb))
+    {
+        return EVP_rc2_ecb();
+    }
+#endif
+
+    return NULL;
 }
 
 const EVP_CIPHER* CryptoNative_EvpRC2Cbc(void)
 {
     // No error queue impact.
-    return EVP_rc2_cbc();
+#if HAVE_OPENSSL_EVP_RC2_CBC
+    if (API_EXISTS(EVP_rc2_cbc))
+    {
+        return EVP_rc2_cbc();
+    }
+#endif
+
+    return NULL;
```

Comment:
I was expecting

```suggestion
    if (API_EXISTS(EVP_rc2_cbc))
    {
#if !OPENSSL_NO_RC2
        return EVP_rc2_cbc();
#endif
    }

    return NULL;
```

but I can't think of a situation where they'd differ in behavior, so either way is fine.

Reply:
This is consistent with CHACHA right below this one, and I agree that I do not think the order matters.